### PR TITLE
Expert advice ssl v tls

### PR DESCRIPTION
**Pull request by AkiraBrand**
[x] Adds short expert advice article on SSL and TLS

### CODE REVIEW

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+SSL (secure socket layer) and TLS (transport layer security) are a set of protocols that are used when a computer is communicating with an external server to get information.  The protocols ensure secure communication over the web by encrypting and securing any information sent over the world wide web.   
+
+*So what’s the difference between the two?*
+
+There isn’t a difference, actually.  TLS actually replaced SSL, but since SSL has been around for longer, people still refer to TLS as SSL. Confused yet?  Read on!
+
+*When do you use SSL? I mean TLS, I mean..wait, what?*
+
+You use TLS, formally known as SSL, mainly when you send data between a client and a server.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:24]:
```suggestion
You use TLS, formerly known as SSL, mainly when you send data between a client and a server.
```

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+In this blog, you’ll learn the difference between SSL and TLS, when they are used, and why you should use them.
+
+*What are SSL and TLS?*
+
+SSL (secure socket layer) and TLS (transport layer security) are a set of protocols that are used when a computer is communicating with an external server to get information.  The protocols ensure secure communication over the web by encrypting and securing any information sent over the world wide web.   
+
+*So what’s the difference between the two?*
+
+There isn’t a difference, actually.  TLS actually replaced SSL, but since SSL has been around for longer, people still refer to TLS as SSL. Confused yet?  Read on!
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:20]: Please use one space between sentences.

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+You use TLS, formally known as SSL, mainly when you send data between a client and a server.
+
+*Ok, I get it. TLS used to be SSL. But how does it work?*
+
+SSL and TLS work by establishing a handshake protocol between a client and a server.  Once that handshake protocol is established, the client and server decide on a secret key that will be used to encrypt and decrypt all communications between them.
+
+In this way, an eavesdropper cannot read or manipulate any of the data sent between the client and the server, they can only see the endpoints.
+
+Now, the overall process of this handshake protocol looks like this:
+
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:33]:
```suggestion
```

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+1. The client sends a “clienthello” message to the server.  This message includes information like SSL/TLS version, and the cryptographic algorithm and data compression methods that the client can support.  
+2. The server responds with a “serverhello” message. This message includes cryptographic algorithms that the server has chosen from the list provided by the client. In this way, an algorithm that both sides support is chosen. The server also sends a session Id, its digital certificate, and its public key.
+3. The client contacts the server’s certificate authority to verify the validity of the server’s certificate. This confirms the authenticity of the web server.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:37]: "The client contacts the server’s certificate authority"

Is this true? I thought that it would look at the list of certificates it knows about and see if the certificate was signed by one of them.

 http://certificate.fyicenter.com/925_Trusted_Root_Certification_Authorities_in_Google_Chrome.html for example.

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+4. A secret key is exchanged.  The client sends a shared secret key to be used between the client and the server.  The secret key is encrypted with the server’s public key.
+5. The client sends a “finished” message to the server, which is encrypted with the secret key.  This confirms that the client section of the handshake protocol is completed.
+6. The server responds with a “finished” message, encrypted with the secret key. This indicates the server part of the handshake protocol is finished.  
+7. The client and server now exchange messages that are symmetrically encrypted with the shared secret key.
+
+{% include _image.liquid src="/site/_diagrams/learn/expert-advice/ssl-vs-tls/ssl-handshake.plantuml" alt="SSL/TLS handshake protocol" class="img-fluid" figure=false %}  
+
+Tl;dr, the entire purpose of the handshake is to verify the validity of the server, as well as securely exchange a shared secret key, which is used to encrypt and decrypt messages sent between the client and the server.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:46]:
```suggestion
Tl;dr, the entire purpose of the handshake is to verify the validity of the server, as well as securely exchange a shared secret key, which is used to encrypt and decrypt messages sent between the client and the server. Encryption with a shared secret is much faster, but the handshake with the asymmetric keys is required to safely share that secret.
```

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+*Why should I care?*
+
+Imagine you are sending a very important message to someone through the mail, that if it was intercepted and read along the way, would have disastrous consequences for your life.  You would need some way of coding the message so if someone finds it, they can’t understand it (encryption) and putting it in an envelope so that the message is secure (securing the communication).  In the same way, information sent over the internet is assumed to need encryption and security, because all kinds of data needs to be sent over the web: credit card numbers, secret love notes, health information, you name it. It all needs to be encrypted and secured, and the way that happens currently is with SSL/TLS.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:50]:
```suggestion
Imagine you are sending a very important message to someone through the mail, that if it was intercepted and read along the way, would have disastrous consequences for your life.  You would need some way of coding the message so if someone finds it, they can’t understand it (encryption) and putting it in an envelope so that the message is secure (securing the communication).  In the same way, information sent over the internet needs encryption and security, because all kinds of data is sent over the web: credit card numbers, secret love notes, health information, you name it. It all needs to be encrypted and secured, and the way that happens currently is typically SSL/TLS.
```

**Review by mooreds — CHANGES_REQUESTED** [review]
A few changes.

**Comment by mooreds** [comment]
Also, don't forget to update the index.html of the security subdirectory.

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+7. The client and server now exchange messages that are symmetrically encrypted with the shared secret key.
+
+
+{% plantuml source: _diagrams/learn/expert-advice/ssl-vs-tls/ssl-handshake.plantuml, alt: "The SSL/TLS handshake protocol." %}
+
+Tl;dr, the entire purpose of the handshake is to verify the validity of the server, as well as securely exchange a shared secret key, which is used to encrypt and decrypt messages sent between the client and the server.
+
+*Why should I care?*
+
+Imagine you are sending a very important message to someone through the mail, that if it was intercepted and read along the way, would have disastrous consequences for your life.  You would need some way of coding the message so if someone finds it, they can’t understand it (encryption) and putting it in an envelope so that the message is secure (securing the communication).  In the same way, information sent over the internet is assumed to need encryption and security, because all kinds of data needs to be sent over the web: credit card numbers, secret love notes, health information, you name it. It all needs to be encrypted and secured, and the way that happens currently is with SSL/TLS.
```
Comment (robotdan) [inline site/learn/expert-advice/security/ssl-vs-tls.md:50]: > Imagine you are sending a very important message to someone through the mail, that if it was intercepted and read along the way, would have disastrous consequences for your life.

Is this a run-on sentence? (I'm not a grammar expert)

Suggestion:

> Imagine you are sending a very important message to someone through the mail. If this email was intercepted or read along the way, it would have disastrous consequences for your life. 

Or 

> Imagine you are sending a very important message to someone through the mail, that if  intercepted or read along the way, would have disastrous consequences for your life.

**Review by robotdan — COMMENTED** [review]
Is this advice or just a kb doc? 

Maybe we already have a mix - but I think most of our expert advice docs are things that devs need to know to implement FusionAuth or auth things, caveats and pitfalls.  

This is more of a "the more you know" , or a "understanding TLS" which is great,  but does it belong in expert advice? Do we have any other categories to place stuff like this?

**Comment by mooreds** [comment]
@robotdan I think this is the start of a new expert advice section, but we don't really have the perfect place for this kind of introductory, yet domain specific content. I propose we leave this in the security section until we have 3-4 pieces and the new website is live, then we can break the pieces out into a 'bits and bytes' section.

It could also be a blog post, but seems to have more staying power than that venue. It shouldn't live in docs. So expert advice seemed to be the best place for it.

File: site/_diagrams/learn/expert-advice/ssl-vs-tls/ssl-handshake.plantuml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+@startuml
+  !include ../../../_config.plantuml
+
+participant "Client" as cl
+participant "Server" as svr
+participant "Certificate Authorithy" as ca
```
Comment (mooreds) [inline site/_diagrams/learn/expert-advice/ssl-vs-tls/ssl-handshake.plantuml:6]: "Authorithy" is misspelled.

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
+---
+layout: advice
+title: SSL vs. TLS
+description: In this blog, you’ll learn the difference between SSL and TLS, how they are used, and why you should use them.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:4]: Not a blog. Maybe "post" or "article"

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+category: Security
+date: 2021-11-12
+dateModified: 2021-11-12
+---
+
+In this blog, you’ll learn the difference between SSL and TLS, when they are used, and why you should use them.
+
+*What are SSL and TLS?*
+
+SSL (secure socket layer) and TLS (transport layer security) are a set of protocols that are used when a computer is communicating with an external server to get information. The protocols ensure secure communication over the web by encrypting and securing any information sent over the world wide web.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:16]:
```suggestion
SSL (secure socket layer) and TLS (transport layer security) are a set of protocols that are used when a computer is communicating with an external server to get information. The protocols ensure secure communication over the web by encrypting and securing any information sent with the HTTP protocol.
```

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+In this blog, you’ll learn the difference between SSL and TLS, when they are used, and why you should use them.
+
+*What are SSL and TLS?*
+
+SSL (secure socket layer) and TLS (transport layer security) are a set of protocols that are used when a computer is communicating with an external server to get information. The protocols ensure secure communication over the web by encrypting and securing any information sent over the world wide web.   
+
+*So what’s the difference between the two?*
+
+There isn’t a difference, actually. TLS replaced SSL, but since SSL has been around for longer, people still refer to TLS as SSL. Confused yet? Read on!
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:20]: I also am confused. You are saying they are the same, but one replaced the other? You might want to say "there isn't a functional difference" or something similar.

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+Now, the overall process of this handshake protocol looks like this:
+
+
+1. The client sends a “clienthello” message to the server. This message includes information like SSL/TLS version, and the cryptographic algorithm and data compression methods that the client can support.  
+2. The server responds with a “serverhello” message. This message includes cryptographic algorithms that the server has chosen from the list provided by the client. In this way, an algorithm that both sides support is chosen. The server also sends a session Id, its digital certificate, and its public key.
+3. The client verifies the validity of the server’s certificate with the entity that issued it. It does so by first checking to ensure that the server's certificate is not expired and the domain name or IP address on the certificate matches the server's information. Then, the client attempts to verify that the server's certificate has been signed by the certificate authority who authorized it. It does so by verifying that the signature on the certificate has been signed by the certificate authority's public key.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:37]: What is a certificate authority?

File: site/learn/expert-advice/security/ssl-vs-tls.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+8. The client and server now exchange messages that are symmetrically encrypted with the shared secret key.
+
+
+{% plantuml source: _diagrams/learn/expert-advice/ssl-vs-tls/ssl-handshake.plantuml, alt: "The SSL/TLS handshake protocol." %}
+
+Tl;dr, the entire purpose of the handshake is to verify the validity of the server. The handshake also securely exchanges a shared secret key, which is used to encrypt and decrypt messages sent between the client and the server.
+
+*Why should I care?*
+
+Imagine you are sending a very important message to someone through the mail, that if it was intercepted and read along the way, would have disastrous consequences for your life.  You would need some way of coding the message so if someone finds it, they can’t understand it (encryption) and putting it in an envelope so that the message is secure (securing the communication).  In the same way, information sent over the internet is assumed to need encryption and security, because all kinds of data needs to be sent over the web: credit card numbers, secret love notes, health information, you name it. It all needs to be encrypted and secured, and the way that happens currently is with SSL/TLS.
```
Comment (mooreds) [inline site/learn/expert-advice/security/ssl-vs-tls.md:51]: Does it make sense to have this be at the start? "why should you care" is usually a great way to start a post.

**Review by mooreds — CHANGES_REQUESTED** [review]

**Comment by jobannon** [comment]
@mooreds Is this dead doc?  Or something to be merged at a later time?

**Comment by mooreds** [comment]
Might be a nice blog post. I should see if I can pull this in in the next few weeks.

**Comment by mooreds** [comment]
Closing.
